lib/index_documents.rb

```diff
@@ -14,20 +14,28 @@ class IndexDocuments
   }
 
   def process(message)
-    if publishing_app_migrated?(message)
-      raw_links = links_from_payload(message)
-
-      unless raw_links.empty?
-        links = rummager_fields_from_links(raw_links)
-        base_path = document_base_path(message)
-        update_index(base_path, links)
-      end
-    end
-
+    $stdout.puts "Processing message: #{message.payload.inspect}"
+    index_links_from_message(message)
     message.ack
+  rescue GdsApi::HTTPServerError => e
+    $stderr.puts "An error occurred!"
+    $stderr.puts e.inspect
+    message.retry
```

Comment:
Aha, I see. Thanks! Couple of questions:

- The problem is that the worker doesn't seem to restart after it exits. I've had to manually do `sudo service rummager-publishing-api-document-indexer-procfile-worker start`.
- @rboulton writes "but would also trigger alerting if a message was failing too frequently (because the process would restart frequently, too)". This alerting hasn't happened as far as I know.
- My reasoning was that since `HTTPServerError` (5XX) is "expected" because publishing-api is sometimes down, we should handle that explicitly. Wdyt?
- Doesn't relying on upstart cause the exception-raising message to always be in front of the queue? Retry-ing will put it at the back of the queue, so that new messages coming in can still be processed. Is this correct?

Perhaps we should have a quick discussion with @rboulton about failure states, logging and monitoring in the queue. I've had a lot of trouble getting visibility on the problems in the last week.